cmd/controller-manager/main.go

```diff
@@ -192,3 +208,26 @@ func main() {
 	}
 	select {}
 }
+
+func parseControllers(controllers []string) (sets.Set[string], error) {
+	if len(controllers) == 0 {
+		return nil, nil
+	}
+
+	for _, name := range controllers {
+		if strings.ToLower(name) != name {
+			return nil, fmt.Errorf("upper case letters not allowed for controller name %q", name)
+		}
+	}
+
+	return sets.New(controllers...), nil
+}
+
+// shouldStartController checks if the given controller should be started,
+// empty controllers is equivalent to start all.
+func shouldStartController(controller string, controllers sets.Set[string]) bool {
+	if len(controllers) == 0 {
+		return true
+	}
+	return controllers.Has(controller)
+}
```

Comment:
FYI, I think we do not need to write functions here. Just use `sets.New`, `sets.IsSubsset`, and `sets.Has` directly in the `main` func.